### PR TITLE
Remove empty recordings in filtering jobs

### PR DESCRIPTION
**Pull request by Icemole**
This PR adds behavior to remove recordings that are left empty due to the filtering performed. It affects the `FilterCorpusBySegmentDurationJob` and the `FilterCorpusRemoveUnknownWordSegmentsJob`.

### CODE REVIEW

**Review by JackTemaki — APPROVED** [review]
If you want you could move the code into a separate private function in the file outside of the jobs, just to save on redundant code. Otherwise ok

File: corpus/filter.py [hunk 1]
```diff
@@ -321,21 +325,50 @@ def unknown_filter(corpus: corpus.Corpus, recording: corpus.Recording, segment:
...
+    """
+
+    __sis_hash_exclude__ = {"delete_empty_recordings": False}
+
+    def __init__(
+        self,
+        bliss_corpus: Path,
+        min_duration: float = 0.1,
+        max_duration: float = 120.0,
+        delete_empty_recordings: bool = True,
```
Comment (michelwi) [inline corpus/filter.py:356]:
```suggestion
        delete_empty_recordings: bool = False,
```
while i agree that `True` should be used by default, if we change the actual default value to `True`, then this breaks the hashes of all existing setups that used this job and did not specify a value.

**Review by michelwi — CHANGES_REQUESTED** [review]

**Comment by Icemole** [comment]
[This chunk of code](https://github.com/rwth-i6/i6_core/blob/249d34f6236160cfa5b62c64d30652626e6b3a95/corpus/filter.py#L255) is also a bit redundant now, but if I adapted it to the current structure there would be two loops over all_recordings. The theoretical time complexity is the same, but practically we need more time to retrieve from main memory and so on... Let me know if you want this shortened as well.

File: corpus/filter.py (hunk 1 above)
Comment (Icemole) [reply corpus/filter.py:366]: Thanks, I meant to set to False for hash compatibility purposes but this slipped past me.

**Comment by michelwi** [comment]
> This chunk of code is also a bit redundant now

I vote for readability / consistence over minuscule time improvements.

**Review by michelwi — APPROVED** [review]

File: corpus/filter.py (hunk 1 above)
Comment (michelwi) [reply corpus/filter.py]: As you see, the AppTek hashes pipeline had also identified this problem. 👍

**Comment by Icemole** [comment]
Would this be safe for merging now?

**Pull request merged**